server/src/main/java/org/opensearch/search/pipeline/Pipeline.java

```diff
@@ -111,7 +138,17 @@ private static <T extends Processor> List<T> readProcessors(
                 processors.add(processorFactories.get(type).create(processorFactories, tag, description, config));
             }
         }
-        return Collections.unmodifiableList(processors);
+        return processors;
+    }
+
+    List<Processor> flattenAllProcessors() {
+        List<Processor> allProcessors = new ArrayList<>(
+            searchRequestProcessors.size() + searchResponseProcessors.size() + searchPhaseResultsProcessors.size()
+        );
+        allProcessors.addAll(searchRequestProcessors);
+        allProcessors.addAll(searchPhaseResultsProcessors);
+        allProcessors.addAll(searchResponseProcessors);
+        return allProcessors;
```

Comment:
Oh -- I think these changes may be the result of an old merge conflict.

IIRC, I intentionally made the return value of `readProcessors` unmodifiable and I removed `flattenAllProcessors`, because it was unused.

I'll delete them and push that change.